File: teste_python/app/retangulo.py
class Retangulo:
    def __init__(self):
        self.largura = 0
        self.altura = 0

    def definirDimensoes(self, largura, altura):
        if largura > 0 and altura > 0:
            self.largura = largura
            self.altura = altura
            return True
        else:
            return False
        
    def calcularArea(self):
        return self.largura * self.altura
    
    def calcularPerimetro(self):
        return 2 * (self.largura + self.altura)
    

File: teste_js/retangulo.js
const Retangulo = {
    largura: 0,
    altura: 0,

    dimensoes: (largura, altura) => {

        if (largura > 0 && altura > 0) {
            this.largura = largura
            this.altura = altura
            return true
        } else {
            return false
        }
    },

    area: () => {
        return this.largura * this.altura
    },

    perimetro: () => {
        return 2 * (this.largura + this.altura)
    }
}


module.exports = Retangulo  